### PR TITLE
set_usr_logger - tuna/utils/logger.py

### PR DESCRIPTION
**Pull request by technicalgrp89**
set_logger - local function to worker_interface.py is moved to logger.py
renamed as set_usr_logger

### CODE REVIEW

File: tuna/utils/logger.py [hunk 1]
```diff
@@ -30,28 +30,48 @@
...
+                 add_streamhandler: bool = True,
+                 add_filehandler: bool = False) -> logging.Logger:
   """std setup for tuna logger"""
-  log_level = os.environ.get('TUNA_LOGLEVEL', 'INFO').upper()
+  log_level: str = os.environ.get('TUNA_LOGLEVEL', 'INFO').upper()
   logging.basicConfig(level=log_level)
-  logger = logging.getLogger(logger_name)
-  log_file = os.path.join(TUNA_LOG_DIR, logger_name + ".log")
-  formatter = logging.Formatter(
+  logger: logging.Logger = logging.getLogger(logger_name)
```
Comment (msaudulhassan) [inline tuna/utils/logger.py]: a bit of a edge case, but if `logger_name==""`, then `logging.getLogger(logger_name)` would return a `logging.RootLogger`

File: tuna/utils/logger.py [hunk 2]
```diff
@@ -30,28 +30,48 @@
...
     logger.addHandler(stream_handler)
 
   logger.setLevel(log_level.upper() if log_level else logging.DEBUG)
   return logger
+
+
+def set_usr_logger(logger_name: str) -> logging.Logger:
+  """Build logger with user defined name"""
+  log_level: str = os.environ.get('TUNA_WOKER_INTERFACE_LOGLEVEL', "INFO")
+  lgr: logging.Logger = logging.getLogger(logger_name)
```
Comment (msaudulhassan) [inline tuna/utils/logger.py]: same

File: tuna/worker_interface.py [hunk 3]
```diff
@@ -92,6 +93,8 @@ def __init__(self, **kwargs):
     self.label = None
     self.session_id = None
 
+    self.logger: logging.Logger
```
Comment (msaudulhassan) [inline tuna/worker_interface.py:96]: I wonder why this is needed?

File: tuna/utils/logger.py [hunk 4]
```diff
@@ -30,28 +30,48 @@
...
     stream_handler.setFormatter(formatter)
     stream_handler.setLevel(logging.INFO)
     logger.addHandler(stream_handler)
 
   logger.setLevel(log_level.upper() if log_level else logging.DEBUG)
   return logger
+
+
+def set_usr_logger(logger_name: str) -> logging.Logger:
+  """Build logger with user defined name"""
```
Comment (msaudulhassan) [inline tuna/utils/logger.py]: So both `set_usr_logger` and `setup_logger` take in a logger_name and create a logger with that name. Maybe update the docstring to better reflect exactly how set_usr_logger differs from setup_logger

**Review by msaudulhassan — CHANGES_REQUESTED** [review]

File: tuna/utils/logger.py (hunk 4 above)
Comment (technicalgrp89) [reply tuna/utils/logger.py:62]: log_level: str = os.environ.get('TUNA_WOKER_INTERFACE_LOGLEVEL', "INFO")
Have added this already. this differenctiates the logger type

**Review by msaudulhassan — APPROVED** [review]

**Review by cderb — APPROVED** [review]

**Pull request merged**